src/utils.js

```diff
@@ -0,0 +1,85 @@
+import tmp from "tmp";
+import request from "request";
+import url from "url";
+import extract from "extract-zip";
+import fs from "fs";
+import { UI_STORE_FOLDER, ADMIN_PORT, HAPP_PORT } from "./constants";
+import { AdminWebsocket } from "@holochain/conductor-api";
+
+// Download from url to tmp file
+// return tmp file path
+const downloadFile = async (downloadUrl) => {
+    const fileName = tmp.tmpNameSync();
+    let file = fs.createWriteStream(fileName);
+
+    // Clean up url
+    let urlObj = new URL(downloadUrl);
+    urlObj.protocol = "https";
+    downloadUrl = urlObj.toString();
+
+    return new Promise((resolve, reject) => {
+        let stream = request({
+            uri: downloadUrl
+        })
+        .pipe(file)
+        .on('finish', () => {
+            console.log(`Downloaded file from ${downloadUrl} to ${fileName}`);
+            resolve(fileName);
+        })
+        .on('error', (error) => {
+            reject(error);
+        })
+    })
+}
+
+export const installDna = async (happ, agentPubKey) => {
```

Comment:
```suggestion
// NOTE: this code assumes a single DNA per hApp.  This will need to be updated when the hApp bundle
// spec is completed, and the hosted-happ config Yaml file will also need to be likewise updated
export const installDna = async (happ, agentPubKey) => {
```